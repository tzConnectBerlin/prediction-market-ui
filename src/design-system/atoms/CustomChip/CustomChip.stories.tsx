import { Story, Meta } from '@storybook/react/types-6-0';
import { RiRefreshLine } from 'react-icons/ri';
import { ChipProps } from '@material-ui/core';
import { CustomChip } from './CustomChip';

export default {
  title: 'Atom/CustomChip',
  component: CustomChip,
} as Meta;

const handleClick = (e: any) => {
  console.log(e.target.outerText);
};

const Template: Story<ChipProps> = (args) => <CustomChip {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: 'test label',
};

export const iconChip = Template.bind({});
iconChip.args = {
  label: 'test label',
  icon: <RiRefreshLine />,
};

export const ClickableChip = Template.bind({});
ClickableChip.args = {
  label: 'clickable',
  icon: <RiRefreshLine />,
  clickable: true,
  onClick: handleClick,
};

export const CustomColor = Template.bind({});
CustomColor.args = {
  label: 'test label',
  color: 'primary',
};

export const OutLine = Template.bind({});
OutLine.args = {
  label: 'test label',
  variant: 'outlined',
  color: 'primary',
};

export const Disabled = Template.bind({});
Disabled.args = {
  label: 'Disabled label',
  disabled: true,
};
