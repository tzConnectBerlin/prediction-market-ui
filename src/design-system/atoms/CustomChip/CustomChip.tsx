import * as React from 'react';
import styled from '@emotion/styled';
import { Chip, ChipProps, useTheme, Theme } from '@material-ui/core';

interface StyledProps {
  theme: Theme;
}

const StyledChip = styled(Chip)<StyledProps>`
  &.xs {
    font-size: ${({ theme }) => theme.spacing(3 / 2)};
    height: ${({ theme }) => theme.spacing(5 / 2)};
  }

  &.sm {
    font-size: ${({ theme }) => theme.spacing(3 / 2)};
    height: ${({ theme }) => theme.spacing(3)};
  }

  &.md {
    font-size: ${({ theme }) => theme.spacing(2)};
    height: ${({ theme }) => theme.spacing(10 / 3)};
    padding: 0 ${({ theme }) => theme.spacing(3 / 2)};
  }

  &.lg {
    font-size: ${({ theme }) => theme.spacing(5 / 2)};
    height: ${({ theme }) => theme.spacing(4)};
    padding: 0 ${({ theme }) => theme.spacing(2)};
  }
`;

export interface CustomChipProps extends ChipProps {
  chipSize?: 'xs' | 'sm' | 'md' | 'lg';
}

const defaultSize = 'xs';
const defaultColor = 'secondary';

export const CustomChip: React.FC<CustomChipProps> = ({
  chipSize = defaultSize,
  color = defaultColor,
  ...rest
}) => {
  const theme = useTheme();
  return <StyledChip color={color} className={chipSize} {...rest} theme={theme} />;
};
