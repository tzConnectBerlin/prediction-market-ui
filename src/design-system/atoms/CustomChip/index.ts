export { CustomChip, CustomChip as default } from './CustomChip';
