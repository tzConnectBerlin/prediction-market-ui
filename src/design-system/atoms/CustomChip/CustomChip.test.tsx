import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { RiRefreshLine } from 'react-icons/ri';
import { CustomChip } from './CustomChip';

describe('Snapshot testing Chip Component', () => {
  it('renders correctly with small size and secondary color', () => {
    const Chip = renderer.create(<CustomChip label="Chip" />).toJSON();
    expect(Chip).toMatchSnapshot();
  });

  it('renders correctly with icon', () => {
    const Chip = renderer.create(<CustomChip label="Chip" icon={<RiRefreshLine />} />).toJSON();
    expect(Chip).toMatchSnapshot();
  });

  it('renders correctly with outline border', () => {
    const Chip = renderer
      .create(<CustomChip label="Chip" color="primary" variant="outlined" />)
      .toJSON();
    expect(Chip).toMatchSnapshot();
  });

  it('renders correctly disbled mode', () => {
    const Chip = renderer.create(<CustomChip label="Chip" disabled />).toJSON();
    expect(Chip).toMatchSnapshot();
  });
});

describe('Element testing Chip Component', () => {
  it('render correctly with Custom label', async () => {
    const { getByText } = render(<CustomChip label="Custom chip" />);
    expect(getByText(/Custom chip/i)).toBeInTheDocument();
  });
});
