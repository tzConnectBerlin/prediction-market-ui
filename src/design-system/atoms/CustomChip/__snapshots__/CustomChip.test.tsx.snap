// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Chip Component renders correctly disbled mode 1`] = `
.emotion-0 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: #fff;
  background-color: #9c27b0;
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  color: inherit;
  margin-left: 5px;
  margin-right: -6px;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(255, 255, 255, 0.7);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0 .MuiChip-deleteIcon:hover,
.emotion-0 .MuiChip-deleteIcon:active {
  color: #fff;
}

.emotion-0.xs {
  font-size: 12px;
  height: 20px;
}

.emotion-0.sm {
  font-size: 12px;
  height: 24px;
}

.emotion-0.md {
  font-size: 16px;
  height: 26.666666666666668px;
  padding: 0 12px;
}

.emotion-0.lg {
  font-size: 20px;
  height: 32px;
  padding: 0 16px;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

<div
  className="MuiChip-root MuiChip-filled Mui-disabled MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-filledSecondary xs emotion-0"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label MuiChip-labelMedium emotion-1"
  >
    Chip
  </span>
</div>
`;

exports[`Snapshot testing Chip Component renders correctly with icon 1`] = `
.emotion-0 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: #fff;
  background-color: #9c27b0;
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  color: inherit;
  margin-left: 5px;
  margin-right: -6px;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(255, 255, 255, 0.7);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0 .MuiChip-deleteIcon:hover,
.emotion-0 .MuiChip-deleteIcon:active {
  color: #fff;
}

.emotion-0.xs {
  font-size: 12px;
  height: 20px;
}

.emotion-0.sm {
  font-size: 12px;
  height: 24px;
}

.emotion-0.md {
  font-size: 16px;
  height: 26.666666666666668px;
  padding: 0 12px;
}

.emotion-0.lg {
  font-size: 20px;
  height: 32px;
  padding: 0 16px;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

<div
  className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-filledSecondary xs emotion-0"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <svg
    className="MuiChip-icon MuiChip-iconMedium MuiChip-iconColorSecondary"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M5.463 4.433A9.961 9.961 0 0 1 12 2c5.523 0 10 4.477 10 10 0 2.136-.67 4.116-1.81 5.74L17 12h3A8 8 0 0 0 6.46 6.228l-.997-1.795zm13.074 15.134A9.961 9.961 0 0 1 12 22C6.477 22 2 17.523 2 12c0-2.136.67-4.116 1.81-5.74L7 12H4a8 8 0 0 0 13.54 5.772l.997 1.795z"
      />
    </g>
  </svg>
  <span
    className="MuiChip-label MuiChip-labelMedium emotion-1"
  >
    Chip
  </span>
</div>
`;

exports[`Snapshot testing Chip Component renders correctly with outline border 1`] = `
.emotion-0 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: #fff;
  background-color: #1976d2;
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  background-color: transparent;
  border: 1px solid rgba(25, 118, 210, 0.7);
  color: #1976d2;
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  color: inherit;
  margin-left: 5px;
  margin-right: -6px;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(255, 255, 255, 0.7);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0 .MuiChip-deleteIcon:hover,
.emotion-0 .MuiChip-deleteIcon:active {
  color: #fff;
}

.emotion-0.MuiChip-clickable:hover {
  background-color: rgba(25, 118, 210, 0.04);
}

.emotion-0.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.12);
}

.emotion-0 .MuiChip-avatar {
  margin-left: 4px;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 2px;
}

.emotion-0 .MuiChip-icon {
  margin-left: 4px;
}

.emotion-0 .MuiChip-iconSmall {
  margin-left: 2px;
}

.emotion-0 .MuiChip-deleteIcon {
  color: rgba(25, 118, 210, 0.7);
}

.emotion-0 .MuiChip-deleteIcon:hover,
.emotion-0 .MuiChip-deleteIcon:active {
  color: #1976d2;
}

.emotion-0 .MuiChip-deleteIconSmall {
  margin-right: 3px;
}

.emotion-0.xs {
  font-size: 12px;
  height: 20px;
}

.emotion-0.sm {
  font-size: 12px;
  height: 24px;
}

.emotion-0.md {
  font-size: 16px;
  height: 26.666666666666668px;
  padding: 0 12px;
}

.emotion-0.lg {
  font-size: 20px;
  height: 32px;
  padding: 0 16px;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

<div
  className="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary xs emotion-0"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label MuiChip-labelMedium emotion-1"
  >
    Chip
  </span>
</div>
`;

exports[`Snapshot testing Chip Component renders correctly with small size and secondary color 1`] = `
.emotion-0 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: #fff;
  background-color: #9c27b0;
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-0 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-0 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-0 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-0 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-0 .MuiChip-icon {
  color: inherit;
  margin-left: 5px;
  margin-right: -6px;
}

.emotion-0 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(255, 255, 255, 0.7);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-0 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-0 .MuiChip-deleteIcon:hover,
.emotion-0 .MuiChip-deleteIcon:active {
  color: #fff;
}

.emotion-0.xs {
  font-size: 12px;
  height: 20px;
}

.emotion-0.sm {
  font-size: 12px;
  height: 24px;
}

.emotion-0.md {
  font-size: 16px;
  height: 26.666666666666668px;
  padding: 0 12px;
}

.emotion-0.lg {
  font-size: 20px;
  height: 32px;
  padding: 0 16px;
}

.emotion-1 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

<div
  className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-filledSecondary xs emotion-0"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label MuiChip-labelMedium emotion-1"
  >
    Chip
  </span>
</div>
`;
